### ISSUE TITLE
Knowledge Distillation: Scaling of the distillation loss

### ISSUE THREAD

**Issue by katyagovorkova** [keras-team/keras-io#869]
According to https://arxiv.org/pdf/1503.02531.pdf it is recommended to scale the distillation loss with T^2:

> Since the magnitudes of the gradients produced by the soft targets scale as 1/T 2
> it is important to multiply them by T^2 when using both hard and soft targets. This ensures that the
> relative contributions of the hard and soft targets remain roughly unchanged if the temperature used
> for distillation is changed while experimenting with meta-parameters.

There is currently **no** T^2 scaling in the [Keras Vanilla Knowledge DIstillation example](https://keras.io/examples/vision/knowledge_distillation/) 

To reflect on the content of the paper, I would suggest including the scaling described in the paper by changing the [line 110](https://github.com/keras-team/keras-io/blob/1caa8696350af88ef40ecef462fd82e429bf7b8a/examples/vision/knowledge_distillation.py#L110) from `)` to `) * self.temperature**2`.
